zio-config/src/main/scala/zio/config/actions/Read.scala

```diff
@@ -15,7 +15,7 @@ object Read {
       report: Ref[ConfigReport]
     ): ZIO[ConfigSource, ReadErrors, (ConfigReport, B)] =
       configuration match {
-        case Config.Pure(value) => report.get.map(t => (t, value))
+        case Config.Empty() => ZIO.fail(ReadErrors(MissingValue("undefined")))
```

Comment:
Empty silently pass the previous computation to next? And not fail?

Reply:
I think empty makes more sense to me than the name  Pure. However we don’t need to fail the read

Reply:
do you mean empty should wrap a value: A ? (like Pure)

Reply:
We could have just 
Config.empty[A] which returns a Config[Option[A]]

During read we can feed in just none. 

And in sequence,  instead of Config.Pure(Nil), we could do Config.empty[A].xmapEither(_.toList)(_.headOption)

I am not sure. Just an idea. It will make sense to user when Config.empty[A] is populating Config with a none value. What do you think ?

Reply:
I use to depend on monoid during these situations. That we can fold over an F, to form a monoidal A.

Reply:
Good idea! Thanks!